File: oldProjectStructure/typescript/P0003/largest-prime-factor/test/Solution.spec.ts
import { Solution } from '../src/Solution';

describe('Helpers', () => {
  test('The largest prime factor of 13195 should be 29', () => {
    const ans = Solution.largestPrimeFactor(13195);
    expect(ans).toBe(29);
  });

  test('The sum of the multiples of 3 and 5 below 1000 should be 233168', () => {
    const ans = Solution.largestPrimeFactor(600851475143);
    expect(ans).toBe(6857);
  });
});

File: oldProjectStructure/haskell/P0003/largest-prime-factors/test/SolutionSpec.hs
module SolutionSpec where

import Solution

import Test.Hspec (Spec, describe, context, it, shouldBe)

-- |Required for auto-discpvery
spec :: Spec
spec =
  describe "Project Euler Problem3 Solution" $ do

    describe "primeFactors 13195" $ do
      it "finds the prime factor of 13195" $ do
         show (primeFactors 13195 []) `shouldBe` "[29,13,7,5]"

    describe "largestPrimeFactor 600851475143" $ do
      it "finds the largest prime factor of 600851475143" $ do
        show (largestPrimeFactor 600851475143) `shouldBe` "6857"